components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/RegistryPolicyDAOImpl.java

```diff
@@ -0,0 +1,1232 @@
+            }
+            PolicyAttributeBuilder policyAttributeBuilder = new PolicyAttributeBuilder();
+            dto.setAttributeDTOs(policyAttributeBuilder.
+                    getPolicyMetaDataFromRegistryProperties(resource.getProperties()));
+            return dto;
+        } catch (RegistryException e) {
+            log.error("Error while loading entitlement policy", e);
+            throw new EntitlementException("Error while loading entitlement policy", e);
+        }
+    }
+
+    /**
+     * Updates the given registry resource.
+     *
+     * @param policy         publishing policy
+     * @param collectionPath registry collection path
+     * @param policyPath     registry resource path
+     * @throws EntitlementException If an error occurs
+     */
+    private void updateResource(PolicyStoreDTO policy, String collectionPath, String policyPath)
+            throws EntitlementException {
+
+        Collection policyCollection;
+        Resource resource;
+
+        try {
+            policyCollection = (Collection) registry.get(collectionPath);
+            resource = registry.get(policyPath);
+
+            if (policy.isSetActive()) {
+                resource.setProperty(PDPConstants.ACTIVE_POLICY, Boolean.toString(policy.isActive()));
+            }
+            if (policy.isSetOrder()) {
+                int order = policy.getPolicyOrder();
+                if (order > 0) {
+                    if (Objects.equals(collectionPath, PDPConstants.ENTITLEMENT_POLICY_PAP)) {
+                        String noOfPolicies = policyCollection.getProperty(PDPConstants.MAX_POLICY_ORDER);
+                        if (noOfPolicies != null && Integer.parseInt(noOfPolicies) < order) {
+                            policyCollection.setProperty(PDPConstants.MAX_POLICY_ORDER, Integer.toString(order));
+                            registry.put(PDPConstants.ENTITLEMENT_POLICY_PAP, policyCollection);
+                        }
+                    }
+                    resource.setProperty(PDPConstants.POLICY_ORDER, Integer.toString(order));
+                }
+            }
+
+            if (policy.isSetOrder() || policy.isSetActive()) {
+                resource.setProperty(PDPConstants.LAST_MODIFIED_TIME, Long.toString(System.currentTimeMillis()));
+                resource.setProperty(PDPConstants.LAST_MODIFIED_USER,
+                        CarbonContext.getThreadLocalCarbonContext().getUsername());
+            }
+
+            registry.put(policyPath, resource);
+
+        } catch (RegistryException e) {
+            log.error("Error while publishing policy", e);
+            throw new EntitlementException("Error while publishing policy", e);
+        }
+    }
+
+    /**
+     * Gets the policy store path
+     *
+     * @return policy store path
+     */
+    private static String getPolicyStorePath() {
+
+        String policyStorePath;
+        policyStorePath = EntitlementServiceComponent.getEntitlementConfig().getEngineProperties().
```

Comment:
engineProperties only contain EntitlementAttributes, but `POLICY_STORE_PATH` is contained in PolicyFinder properties, which was previously populated when the init method was called. Must check if this change is intentional.